File: usaco/beads.py
import math
data = open('beads.in', 'r')
index, s = data.read().split('\n')


def findValues(up, index):
    value = 1
    iteration = 0
    initValue = coolString[index]
    tIndex = index
    while (initValue == 'w'):
        initValue = coolString[tIndex + up]
        tIndex += up
    while (coolString[index + up] == initValue or coolString[index + up] == 'w'
           and iteration < math.floor(index / 2)):
        value += 1
        index += up
        iteration += 1
    return value


maxe = 0
coolString = s * 3
initValue = s[index]

for ite in range(index, 2 * index):
    lVal = findValues(1, ite)
    rVal = findValues(-1, ite)
    cVal = lVal + rVal
    if (cVal > maxe):
        maxe = cVal

if (max > index):
  print(index)
else:
  print(maxe)

File: usaco/beads.cpp
/*
ID: futzipe1
TASK: test
LANG: C++                 
*/
#include <iostream>
#include <fstream>
#include <string>

using namespace std;

findValues(bool left, index)
{
    int value = 1, iteration = 0;
    char initValue = coolString[index];
    int tIndex = index;
    if (left)
    {
        while (initValue == 'w'))
            {
                initValue = coolString[tIndex - 1] tIndex--;
            }
        while (coolString[index - 1] == initValue || coolString[index - 1] == 'w' && iteration < strL / 2)
        {
            value++;
            index--;
            iteration++;
        }
        return value;
    }
    else
    {
        while (initValue == 'w'))
            {
                initValue = coolString[tIndex + 1] tIndex++;
            }
        while (coolString[index + 1] == initValue || coolString[index + 1] == 'w' && iteration < strL / 2)
        {
            value++;
            index++;
            iteration++;
        }
        return value;
    }
}

int main()
{
    catchSpecialCase()

        int max = 0 int index,
            lINdex, rIndex;

    coolString = s + s;
    strL = string.length();
    iterationRange = coolString(strL / 2, strL)

                             stop conditions : 1. if the color is not white and
                         the color is no longer the same 2. if iteration has gone to 1 / 2 string length or
                     end of string

                                 start at
                                     stringlength /
                             2 and
                         go to stringlength * 1.5

                             for (char c : string iterationRange)
    {
        lValue = findValues(left, index)
            rValue = findValues(right, index)
                cValue = lValue + rValue if (cValue > max) max = cValue
    }

    return max;
}

/*

